bullet_train/app/controllers/concerns/account/controllers/base.rb

```diff
@@ -64,7 +64,7 @@ def ensure_onboarding_is_complete_and_set_next_step
   def ensure_onboarding_is_complete
     # This is temporary, but if we've gotten this far and `@team` is loaded, we need to ensure current_team is
     # updated for the checks below. This entire concept of `current_team` is going away soon.
-    current_user.update(current_team: @team) if @team&.persisted?
+    current_user.update(current_team: @team) if @team&.persisted? && current_user.teams.include?(@team)
```

Comment:
Is the intent here that we update the `current_team` when a user visits the `show` action (or similar) of one of their teams?

Reply:
The reason I added this conditional here was because since I allowed users to visit other teams that aren't theirs, it would add the team their visiting as their current team 😬

So I used this as a safeguard to ensure only teams they belong to are updated to `current_team`.